passes/src/Luna/Pass/Evaluation/Interpreter.hs

```diff
@@ -61,6 +62,37 @@ mergeScopes m = localVars %~ Map.union m
 globalLookup :: Name -> Imports -> Maybe LunaValue
 globalLookup n imps = imps ^? importedFunctions . ix n . documentedItem . _Right . value
 
+
+
+newtype ScopeT m a = ScopeT { unScopeT :: State.StateT (IORef LocalScope) m a } deriving (Functor, Applicative, Monad, MonadIO, MonadTrans)
+
+class Monad m => MonadScope m where
+    get :: m LocalScope
+    put :: LocalScope -> m ()
+
+instance MonadIO m => MonadScope (ScopeT m) where
+    get   = ScopeT $ State.get >>= liftIO . readIORef
+    put s = ScopeT $ State.get >>= liftIO . flip writeIORef s
+
+modify :: MonadScope m => (LocalScope -> LocalScope) -> m ()
+modify f = fmap f get >>= put
```

Comment:
Is there a reason for the explicit `fmap` rather than `<$>` here? I'd personally find `f <$> get >>= put` to be slightly more readable, if only because most of the codebase I've seen seems to use `<$>`.

Reply:
Mostly because I hate thinking about operators precedence and this allowed me to avoid it.

Reply:
Fair enough. The style guide doesn't have anything about a preference, so I'd leave it as is unless someone else has a reason not to!